meetings/2022-02-28.md

```diff
@@ -0,0 +1,131 @@
+* AC: This kind of spec organization will make things easier for implementers and people testing implementations. Because there are a number of conceptual entites involved, each having its own responsibilities. Unless you understand each of these components, it is confusing who is responsible for what. The new organization intends at reducing this confusion. KD, since you're looking at implementing an OP your feedback will be very valuable.
+* KD: One of the recent changes made the sequence diagram noticeably more complex. We still have a lot of areas to investigate to figure out what responsibilites our software will have.
+* AC: It's really great that you're asking the questions to the panel.
+* EP: ID token verification is the next step I want to address, without specifying who is doing the verification. Maybe describing a mixin class "OIDC token verifier", and refer to it for different conformance classes
+* NS: it would be useful to have these mixin classes, but it might be interesting to consider whether that verification should happen differently, depending on the party. e.g., certain claims may need to be validated differently. May also want to consider having a primary conformance class for how an ID token should be validated, then other conformance classes could refer to that conformance class.
+* EP: The specific use case we are looking at is the interop Authorization Agent acting as a client to the OP, in which case we want to verify the webid claim as a client. AC, is it something you've met in your experience?
+* AC: Any server code does full validation of the token. If the client makes use of the values of that ID token, it is good to validate the signature. Needing to do full end-to-end validation may depend on the use case.
+* LD: We have a use case where the ID token is parsed on the client side too. We operate our own OP though, so there is a stronger trust relationship. In the case where your client doesn't have that, full validation mùakes sense.
+* KD: in the case of an SPA, shouldn't the ID token always be verified ?
+* AC: There are multiple ways to approach this. Existing patterns are: 1) asking the user where to start (one or several OP), 2) ask the WbID to the user (tricky from a UI perspective). An SPA could store some info in local storage, but it should be always considered as something that may have been tampered with. Then the client may validate the previous value with the user. Another common pattern is to provide a list of common OP, which is only valid when there is a limited number of OPs. 
+* NAS: The solid-client-authn library does signature validation of the ID token
+* EP: Verifying the signature comes form the OIDC spec. Issuer could be validated by the client, and must be by the AS, and the DPop validation doesn't really make sense on the client side since they are the ones holding the private key.
```

Comment:
```suggestion
* EP: Verifying the signature comes from the OIDC spec. Issuer could be validated by the client, and must be by the AS, and the DPop validation doesn't really make sense on the client side since they are the ones holding the private key.
```